signed shift result (0x80000000) sets the sign bit of the shift expression's type ('int') and becomes negative

```
xsimd/include/xsimd/arch/xsimd_sse2.hpp:545:38: error: signed shift result (0x80000000) sets the sign bit of the shift expression's type ('int') and becomes negative [-Werror,-Wshift-sign-overflow]
                    _mm_set1_epi32(1 << 31)));
                                   ~ ^  ~~
```

--- a/include/xsimd/arch/xsimd_avx.hpp
+++ b/include/xsimd/arch/xsimd_avx.hpp
@@ -534,7 +534,7 @@ namespace xsimd
                     _mm256_blendv_ps(_mm256_castsi256_ps(_mm256_cvttps_epi32(self)),
                                      _mm256_xor_ps(
                                          _mm256_castsi256_ps(_mm256_cvttps_epi32(_mm256_sub_ps(self, _mm256_set1_ps(1u << 31)))),
-                                         _mm256_castsi256_ps(_mm256_set1_epi32(1 << 31))),
+                                         _mm256_castsi256_ps(_mm256_set1_epi32(1u << 31))),
                                      _mm256_cmp_ps(self, _mm256_set1_ps(1u << 31), _CMP_GE_OQ)));
             }
         }

--- a/include/xsimd/arch/xsimd_avx512f.hpp
+++ b/include/xsimd/arch/xsimd_avx512f.hpp
@@ -1317,7 +1317,7 @@ namespace xsimd
                 alignas(avx2::alignment()) uint8_t buffer[64];
                 // FIXME: ultra inefficient
                 for (int i = 0; i < 64; ++i)
-                    buffer[i] = cond.data & ((uint64_t)1 << i) ? 0xFF : 0;
+                    buffer[i] = cond.data & (1ull << i) ? 0xFF : 0;
                 __m256i cond_low = batch<uint8_t, avx2>::load_aligned(&buffer[0]);
                 __m256i cond_hi = batch<uint8_t, avx2>::load_aligned(&buffer[32]);
 

--- a/include/xsimd/arch/xsimd_sse2.hpp
+++ b/include/xsimd/arch/xsimd_sse2.hpp
@@ -546,7 +546,7 @@ namespace xsimd
                 __m128 lhs = _mm_castsi128_ps(_mm_cvttps_epi32(self));
                 __m128 rhs = _mm_castsi128_ps(_mm_xor_si128(
                     _mm_cvttps_epi32(_mm_sub_ps(self, _mm_set1_ps(1u << 31))),
-                    _mm_set1_epi32(1 << 31)));
+                    _mm_set1_epi32(1u << 31)));
                 return _mm_castps_si128(_mm_or_ps(_mm_and_ps(mask, rhs), _mm_andnot_ps(mask, lhs)));
             }
 

--- a/include/xsimd/arch/xsimd_sse4_1.hpp
+++ b/include/xsimd/arch/xsimd_sse4_1.hpp
@@ -73,7 +73,7 @@ namespace xsimd
                     _mm_blendv_ps(_mm_castsi128_ps(_mm_cvttps_epi32(self)),
                                   _mm_castsi128_ps(_mm_xor_si128(
                                       _mm_cvttps_epi32(_mm_sub_ps(self, _mm_set1_ps(1u << 31))),
-                                      _mm_set1_epi32(1 << 31))),
+                                      _mm_set1_epi32(1u << 31))),
                                   _mm_cmpge_ps(self, _mm_set1_ps(1u << 31))));
             }
         }

